### ISSUE TITLE
Docs: Consider using "GitHub" instead of "GitHub Repository" as aria-label for GH icon

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#5232]
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

Right now the nav header GitHub icon has `aria-label="GitHub Repository'`. That _generally_ works fine to describe what the icon is. However, users who use software that maps from typing link titles on a keyboard to the actual link might find that non-exact matches don't get mapped as nicely. 

Discussed on stream with @BenDMyers on Twitch: https://www.twitch.tv/videos/1498962341. 

I tried Googling around and couldn't find specific instances of this advice. https://www.w3.org/TR/WCAG20-TECHS/H33.html has general-purpose accessibility docs around link titles.

On the other hand, I can also see an argument for the more verbose title being more helpful? That it provides more context as to the link's purpose?

### Affected URL(s)

\*

**Comment by BenDMyers** [comment]
I'd flag this as very, _very_ minimal priority. What you've got is totally fine, and the recommendation to go to "GitHub" is just for _maximum_ predictability. I definitely wouldn't flag this as a bug.

The use case I gave for this was speech recognition/voice control software. Users who want to click a link or button can say "click _[link/button name]_," such as "click Comment." When it comes to icon links and buttons, sighted voice control users will have to guess at the underlying name — so providing a predictable name is paramount. In my opinion, "GitHub" is the most predictable name, given the link is the GitHub logo.

However, the reason I'm not flagging this as very critical is that voice control software _can_ do some fuzzy matching - specifically if the bit the user guesses is a substring of the link/button's actual name. So if the user says "click GitHub," voice control can interpret that as a command to click the "GitHub repository" link, provided no other link is a better fit. For maximum interoperability, the bit the user says should be at the _beginning_ of the link/button's full name (so if we expect folks to say "GitHub," then the link's name being "GitHub repository" is probably better than it something like "Go to GitHub.")

All this to say that the current link probably is fine (and would not fail any audits), but that shortening the name to just "GitHub" could make it even more predictable and guarantee even more interop. That's the context for the recommendation I gave, but I also think it's totally fine to disregard this.

**Comment by JoshuaKGoldberg** [comment]
Ok cool in that case I do like having the increased specificity - for users who aren't accustomed to a GitHub link in a header and/or footer, it's nice that this one specifies that it's the repo.

Thanks for the explanation @BenDMyers! ❤️
